test/test-google-sheets-api.ts

```diff
@@ -46,37 +81,62 @@ function authorize(credentials: any, callback: any) {
   });
 }
 
+/**
+ * Gets an enviroment variable, but throws an error if it's undefined.
+ */
+function getEnvOrError(name: string): string {
+  const env = process.env[name];
+  if (env === undefined) {
+    throw new Error(
+      `Did not find ${name} in environment. Did you forget to define it in the .env file?`
+    );
+  }
+  return env;
+}
+
+/////////////////////////////////// Tests ////////////////////////////////////
+
 test.cb("it should fetch wordlist from Google Sheets API", (t) => {
   authorize(
     {
-      client_secret: process.env.CLIENT_SECRET,
-      client_id: process.env.CLIENT_ID,
-      redirect_uri: process.env.REDIRECT_URI,
+      client_secret: getEnvOrError("CLIENT_SECRET"),
+      client_id: getEnvOrError("CLIENT_ID"),
+      redirect_uri: process.env.REDIRECT_URI || "http://localhost:5000",
     },
-    (auth: any) => {
+    (auth: Auth) => {
       const sheets = google.sheets({ version: "v4", auth });
-      const spreadsheetId = process.env.GOOGLESHEETS_ID || "";
+      const spreadsheetId = SPREADSHEET_ID;
       sheets.spreadsheets.values.get(
         {
           spreadsheetId,
           range: "A1:B",
         },
-        async (err, res: any) => {
+        async (err, res: unknown) => {
```

Comment:
So this kind of sucks, but TypeScript knows the type is something deep within the `googleapis` library, and the types here are, strictly speaking, not that! So it has to be `unknown` and then cast to an appropriate type.

There's probably a better way of doing this, but it involves importing types from deep within `googleapis` which is also not great 🙃

Reply:
Absolutely! I definitely see the pain here. I personally don't have a better alternative as of now, perhaps someone could bring in a suggestion when they bump into this later.